jobs/integr8ly/ocp3/test-suites/all-tests-executor.yaml

```diff
@@ -143,9 +143,12 @@
                         runTests('sso-user-create-tests')
                     } // stage
 
-                    stage ('Nexus builds test') {
-                        runTests('nexus-builds-test')
-                    } // stage
+                    // run nexus-build test only on PDS
+                    if (CLUSTER_URL.contains("open.redhat.com")) {
```

Comment:
Is it only me who thinks that this should be handled in the test level, not pipeline level. E.g. if there is no Nexus namespace the test file should be executed but the tests inside it should be programmatically skipped.

I think it is better to have our tests standalone as much as possible and pipeline should just trigger them.

Reply:
TBH I am not sure how to handle it nicely in the test source code so if you have some suggestion then I am one ear.

Reply:
what about this:
https://mochajs.org/#pending-tests (search for `this.skip()`) ?

Reply:
It seems if you do that in beforeAll you can skip all the tests (which is what we want if Nexus namespace is not present.

Reply:
OK interesting. This could be a cleaner solution (although I am not sure why the test is marked as "pending"), but on the other hand the stage will be still present in the pipeline, which could be a little bit confusing...but I don't have a strong preference here.

Feel free to close it and assign the issue to yourself :)

Reply:
yeah, I really think it is better solution to do such things on the test level, not pipeline level since tests should be standalone and solve such things themselves.

I also think having the stage always there is less confusing than some stages only being present some times. Jenkins does not handle such situation nicely when displaying history and I think it is better to see the stage always there, so you can have a look and see that test were executed but effectively skipped - with a message why were skipped in the test log

Reply:
Feel free to unassign yourself from the ticket fist ;)